Return 401 in all cases of failure

robsliwi opened the pull request:
In some cases buffalo would return a 500 instead
of a 401 when the authorization fails.
This commit should fix this behavior, see
discussion in Slack:
https://gophers.slack.com/archives/C3MSAFD40/p1543923920092000

markbates commented:
The tests are failing for this, can you please take a look? Thanks.

markbates commented on basicauth.go:
```diff
@@ -10,11 +10,8 @@ import (
 )
 
 var (
-	// ErrNoCreds is returned when no basic auth credentials are defined
```
Removing these constitutes a breaking change.

markbates commented on basicauth_test.go:
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/gobuffalo/buffalo"
 	"github.com/gobuffalo/httptest"
-	"github.com/gobuffalo/mw-basicauth"
+	"github.com/robsliwi/mw-basicauth"
```
Broken import path.

Please read https://github.com/gobuffalo/buffalo/blob/master/.github/CONTRIBUTING.md on how to handle Go PRs.

robsliwi committed:
Return 401 in all cases of failure

In some cases buffalo would return a 500 instead
of a 401 when the authorization fails.
This commit should fix this behavior, see
discussion in Slack:
https://gophers.slack.com/archives/C3MSAFD40/p1543923920092000

robsliwi commented:
Thanks for the contributing link, should have checked this before making the PR :innocent: 
Updated and rebased it, you can have a second look if you want.

sio4 commented on basicauth.go:
```diff
@@ -15,6 +15,10 @@ var (
 
 	// ErrAuthFail is returned when the client fails basic authentication
 	ErrAuthFail = errors.New("invalid basic auth username or password")
+	// ErrUnauthorized is returned in any case the basic authentication fails
```
extra comment?

sio4 commented on go.mod:
```diff
@@ -1,8 +1,11 @@
-module github.com/gobuffalo/mw-basicauth
+module github.com/robsliwi/mw-basicauth
```
it should not be changed.

sio4 reviewed (changes requested):
Thank you for your contribution, and sorry for the late review.

Could you please check the comments and fix them? I will proceed with this PR after that.
If you have no interest in this module anymore, not sure since it is already more than four years ago, just ignore this message. I will work on it soon.

robsliwi commented:
Thanks for picking this up!
Was quite a joy to clone the repo again, set up some Golang-dev-env and get this fixed.

Rebased, pushed, realized two commits, squashed them and all of the requested changes should be included by now :crossed_fingers:

sio4 commented:
Thanks!

sio4 commented:
Just for your information, I made some more changes to the topic and filed PR #7. Just in case you are interested.
